modules/core/src/main/java/org/apache/ignite/internal/processors/authentication/IgniteAuthenticationProcessor.java

```diff
@@ -836,14 +792,11 @@ private void onAuthenticateResponseMessage(UserAuthenticateResponseMessage msg)
      * Local node joined to topology. Discovery cache is available but no discovery custom message are received.
      * Initial user set and initial user operation (received on join) are processed here.
      */
-    public void onLocalJoin() {
-        if (coordinator() == null)
+    private void onLocalJoin() {
+        if (ctx.isDaemon() || ctx.clientDisconnected() || coordinator() == null)
```

Comment:
Can we replace clientDisconnected with ctx.clientNode(), and then remove a check below. Client disconnected looks weird, as there no actions on reconnected, and it's misleading how a node should fill initial user state.

Reply:
Please, see changes related to GridDiscoveryManager#start. My motivation to add this check:
It was called by GridDiscoveryManager previously and to keep IgntieAuthenticationProcessor behavior intact I duplicated it.

It might be weird but It makes review easier. And it works exactly as before.

Reply:
I see, thanks.